src/runtime_src/core/common/query_requests.h

```diff
@@ -715,6 +722,36 @@ struct elf_name : request
   get(const device*, const std::any& req_type) const override = 0;
 };
 
+struct mobilenet : request 
+{
+  enum class type {
+    mobilenet_ifm,
+    mobilenet_param,
+    buffer_sizes
+  };
+
+  static std::string
+  enum_to_str(const type& type)
+  {
+    switch (type) {
+      case type::mobilenet_ifm:
+        return "mobilenet_ifm";
+      case type::mobilenet_param:
+        return "mobilenet_param";
+      case type::buffer_sizes:
+        return "buffer_sizes";
+    }
+    return "unknown";
+  }
+
+  using result_type = std::string;
+  static const key_type key = key_type::mobilenet;
+  static const char* name() { return "mobilenet"; }
+
```

Comment:
Who are using these? Should these be virtual functions in the request class for each derived class to override?

Reply:
@sonals These are integrals pieces of the query infrastructure.  The infrastructure is templated, these cannot be virtual functions.